### PR TITLE
SecGeoLookupDb directive support

### PR DESCRIPTION
**Pull request by bxlxx**
> Thank you for contributing to Coraza WAF, your effort is greatly appreciated
> Before submitting check if what you want to add to `coraza` list meets [quality standards](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#quality-standards) before sending pull request. Thanks!

**Note**: _that go.mod and go.sum can only be modified for tested dependency updates or justified new features._

**Make sure that you've checked the boxes below before you submit PR:**

- [x] My code includes positive and negative tests.
- [x] I have an appropriate description with correct grammar.
- [x] I have read [Contribution guidelines](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#contribution-guidelines), [maintainers note](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#maintainers) and [Quality standard](https://github.com/github.com/corazawaf/coraza/v2sso/coraza-waf/blob/master/CONTRIBUTING.md#quality-standards).

Thanks for your PR :heart:

### CODE REVIEW

**Review by jptosso — COMMENTED** [review]
We should review waf.GeoIPDB and if the numeric operators supports floats

File: operators/geo_lookup_test.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+func Test_geoLookup(t *testing.T) {
+	geoLookupTest := geoLookup{}
+	if err := geoLookupTest.Init(""); err != nil {
+		t.Errorf("Error init geolookup")
+	}
+
+	var err error
+	waf := engine.NewWaf()
+	waf.GeoIPDB, err = geoip2.Open("../testdata/mmdb/GeoLite2-City.mmdb")
```
[the remark this answers is not in the document]
Comment (bxlxx) [reply operators/geo_lookup_test.go:19]: Good idea. I will remove it. Thanks.

**Review by fzipi — COMMENTED** [review]
Looks good!

Do we need to push the testdata to this repo? Can't we do this in the Github Actions PIpeline? I'm worried about making this repo too heavy...

**Comment by fzipi** [comment]
Also, looks like there are some conflicts in the go.mod file 🤔

**Comment by jptosso** [comment]
Can we transform this into a plugin until we find how to manage the databases? It will be an official plugin with official support.

I think we must keep it simple, in the meantime we can use the waf.Config.Get("geoip") interface to store the database for each engine (one plugin per engine)

I have created this as a sample: https://github.com/corazawaf/coraza-geoip

**Comment by bxlxx** [comment]
> Can we transform this into a plugin until we find how to manage the databases? It will be an official plugin with official support.
> 
> I think we must keep it simple, in the meantime we can use the waf.Config.Get("geoip") interface to store the database for each engine (one plugin per engine)
> 
> I have created this as a sample: https://github.com/corazawaf/coraza-geoip

Sure! Thank you very much!

**Pull request closed without merging**